hack/postrelease/tests/test_images.py

```diff
@@ -0,0 +1,208 @@
+import json
+import subprocess
+
+import pytest
+import requests
+import variables
+
+OPERATOR_IMAGE = f"quay.io/tigera/operator:{variables.OPERATOR_VERSION}"
+
+# Architectures we expect to be present in multi-arch image manifests.
+EXPECTED_ARCHS = ["amd64", "arm64", "armv7", "ppc64le", "s390x"]
+
+# Images we expect to exist as part of a Calico release, without
+# a registry assigned.
+EXPECTED_IMAGES = [
+    "calico/node",
+    "calico/ctl",
+    "calico/apiserver",
+    "calico/typha",
+    "calico/cni",
+    "calico/kube-controllers",
+    "calico/upgrade",
+    "calico/windows",
+    "calico/flannel-migration-controller",
+    "calico/dikastes",
+    "calico/pilot-webhook",
+    "calico/pod2daemon-flexvol",
+    "calico/csi",
+]
+
+TAG_SUFFIXES = [
+    "amd64",
+    "arm64",
+    "armv7",
+    "fips-amd64",
+    "fips",
+    "ppc64le",
+    "s390x",
+]
+
+# Images that we exclude from the assertions below.
+EXCLUDED_IMAGES = ["calico/pilot-webhook", "calico/upgrade"]
+OPERATOR_EXCLUDED_IMAGES = EXCLUDED_IMAGES + [
+    "calico/dikastes",
+    "calico/flannel-migration-controller",
+    "calico/ctl",
+    "calico/windows",
+    "calico/csi",
+]
+
+CHECK_IMAGES = [img for img in EXPECTED_IMAGES if img not in EXCLUDED_IMAGES]
+
+# Images that we expect to be published to GCR.
+GCR_IMAGES = [
+    "calico/node",
+    "calico/cni",
+    "calico/typha",
+]
+
+ALL_IMAGES = []
+for image in EXPECTED_IMAGES:
+    for tag_suffix in TAG_SUFFIXES:
+        tag_suffixed = f"{variables.RELEASE_VERSION}-{tag_suffix}"
+        ALL_IMAGES.append((image, tag_suffixed))
+
+print("Found {} separate image tags to test".format(len(ALL_IMAGES)))
+
+
+def request_quay_image(image_name, image_version):
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": "Bearer {}".format(variables.QUAY_TOKEN),
```

Comment:
I think we should be testing without a quay-token.

Reply:
Agree (as above)